test-suite-groovy/build.gradle

```diff
@@ -21,8 +21,3 @@ dependencies {
 tasks.named('test') {
     useJUnitPlatform()
 }
-
-java {
-    sourceCompatibility = JavaVersion.toVersion("1.8")
```

Comment:
should we set: 

```
java {
    sourceCompatibility = JavaVersion.toVersion("17")
    targetCompatibility = JavaVersion.toVersion("17")
}
```

Create a test-base in buildScript to avoid repetition. 

ping @melix

Reply:
I have created a PR https://github.com/micronaut-projects/micronaut-email/pull/135